### ISSUE TITLE
Update RPC system_dryRun to return emitted events as well

### ISSUE THREAD

**Issue by xlc** [paritytech/substrate#7555]
This will be helpful to figure what what will happen before submitting a transaction.

May need to make a new one to avoid breaking change on return parameter?

At later stage should integrate with state tracing so we can learn more about what exactly will happen.

**Comment by kianenigma** [comment]
First, we should move the system RPC to also be just a normal runtime api callabale by `state_call`.

Then, we should also add a new variant that doesn't accept an extrinsic, but instead a call and an origin. This should be really helpful in testing gov proposals for PJS apps.

**Comment by xlc** [comment]
We are currently using [Chopsticks](http://github.com/AcalaNetwork/chopsticks) to test governance proposals and it should work with Kusama/Polkadot as well.

Eventually some parts of Chopsticks could be compiled into wasm and runnable in browser, and then we will be able to use this to dry run extrinsics instead of asking a node to do it, which could be a DoS vector for fullnode.

**Comment by kianenigma** [comment]
Yeah, but I guess in the meantime we can also add this. A webpage we can technically already run a smaldot node, and it can be configured to respond to unsafe RPC calls only locally. Or, there are other ways to whitelist origins for unsafe RPCs etc.

**Comment by xlc** [comment]
One of the goal of Chopsticks is to make it a library that can be run in browser. Then it can be used to simulate transactions within browser and get all the changed storages.

**Comment by kianenigma** [comment]
Indeed, but I am asking what we can do now to make it easy to dry-run a `Call` in polkadot js apps. I am personally in favor of it.

**Comment by xlc** [comment]
We could see if we can have this feature backed into polkadot.js or substrate-connect
